core/src/main/java/org/mapstruct/BeanMapping.java

```diff
@@ -140,6 +140,15 @@ NullValuePropertyMappingStrategy nullValuePropertyMappingStrategy()
      */
     String[] ignoreUnmappedSourceProperties() default {};
 
+    /**
+     * How unmapped properties of the target type of a mapping should be
+     * reported. The method overrides an unmappedTargetPolicy set by {@link Mapper#unmappedSourcePolicy()} or
+     * {@link MapperConfig#unmappedTargetPolicy() }
```

Comment:
More general question: this "override" language matches the language in `Mapper.java`, but not the other overrides in this class.  For example:

```java
     * The strategy to be applied when {@code null} is passed as source bean argument value to this bean mapping. If no
     * strategy is configured, the strategy given via {@link MapperConfig#nullValueMappingStrategy()} or
     * {@link Mapper#nullValueMappingStrategy()} will be applied, using {@link NullValueMappingStrategy#RETURN_NULL}
     * by default.
```

and

```java
     * The strategy to be applied when a source bean property is {@code null} or not present. If no strategy is
     * configured, the strategy given via {@link MapperConfig#nullValuePropertyMappingStrategy()} or
     * {@link Mapper#nullValuePropertyMappingStrategy()} will be applied,
     * {@link NullValuePropertyMappingStrategy#SET_TO_NULL} will be used by default.
```

I do think think I prefer the approach you're using here, but it is inconsistent.  Should the other Javadocs in this class be updated to use this same linguistic approach?

Reply:
What is your opinion @chris922?

Reply:
I also think that the new wording is clear. It gives the clear intent of this. I think it makes it also clear that if you set it (even if to the default value), it will override what was in `@Mapper` or `@MapperConfig`.

WDYT @sjaakd?

Reply:
If I interpret @LukeLaz his comment correctly it comes down to 2 things:
1. don't use the `override` term because its meaning in OO languages (like in `Overrides`). 
2. keep phrasing aligned over all places were we have this situation.
I tend to agree with that. 

However, if something is unclear - like explicitly setting the default value has a different effect than leaving it out altogether - we need to add that in other places.

Reply:
@sjaakd My comment was specifically regarding #2.  Namely, there currently exists different phrasing between `BeanMapping.java` and `Mapper.java`, but this PR is using the `Mapper.java` phrasing within `BeanMapping.java`.  I suspect the phrasings between the two can be made similar.  If not, it's probably better to keep the same phrasing within the same class.

That said, I'm a user of the library, not a contributor, so that probably limits the value of my opinion somewhat.

Reply:
Hmmm, on further inspection it seems like phrasing is not aligned already:

```
     * Determines when to include a null check on the source property value of a bean mapping.
     *
     * Can be overridden by the one on {@link MapperConfig}, {@link Mapper}  or {@link Mapping}.
     *
     * @return strategy how to do null checking
     * NullValueCheckStrategy nullValueCheckStrategy() default ON_IMPLICIT_CONVERSION;
```